src/rescue.rs

```diff
@@ -0,0 +1,509 @@
+extern crate num_bigint;
+extern crate franklin_crypto;
+use num_bigint::{BigInt,ToBigInt,Sign};
+use franklin_crypto::circuit::{
+    uint32::UInt32,
+    multieq::MultiEq,
+    boolean::Boolean,
+    num::*,
+    expression::Expression,
+    Assignment
+};
+use franklin_crypto::bellman::{
+    ConstraintSystem, 
+    SynthesisError,
+    pairing::Engine,
+    LinearCombination,
+    Variable,
+    pairing::ff::{
+        Field,
+        PrimeField,
+        PrimeFieldRepr,
+        BitIterator
+    }
+};
+
+pub fn rescue_hash<E, CS>(
```

Comment:
Rescue does not hash booleans, but only allocated numbers

Reply:
So we don't need this function, but the one with vector of allocated numbers instead?